### ISSUE TITLE
Some UI bugs and possible UX improvements

### ISSUE THREAD

**Issue by DumboJetEngine** [opnsense/plugins#1846]
Hello.
Thank you for the great firewall. :)
I really like it.
I have a list of UI bugs/improvements, in case you are interested:

1)
Go to:
Services>Intrusion Detection>Administration>Alerts
The pagination does not work.
E.g., I select a large page size and I see multiple pages shown.
But only the first page has content.
![image](https://user-images.githubusercontent.com/8676903/82726105-abfd3580-9cea-11ea-928a-fe4096b821b1.png)
Also, the timestamps shown on this table should probably be formatted to something more readable by a user. Right now you see things like this: **2020-05-23T08:04:54.384826+0000**.

2)
Go to:
Services>Intrusion Detection>Administration>Download
Write "ET tele" on the filter.
Press the "Download and Update Rules" button at the bottom.
First problem: once the action completes, the filter is not applied to the results anymore.
Extra problem: editing the filter does not always re-filter the results. E.g., when cutting and pasting things on the filter, you have to press enter to force the filter to execute. This is really weird because the first impression that you get is that this would filter automatically when you changed the filter field.
I believe that the filtering is hooked up to the wrong event. It should be the 'change' event normally.

3)
Go to:
Services>Let's Encrypt>Settings>Button "Reset acme client"
Clicking on the button, presents this modal:
![image](https://user-images.githubusercontent.com/8676903/82726232-4cebf080-9ceb-11ea-8745-2b146f7af776.png)
The first time I saw it, I thought that the buttons were wired the wrong way.
Now, I am not sure and I am too afraid to click any of them (although I think I did in the past).
But I think that the trash icon should not be in the Abort button, but on the button that actually deletes data (Continue). Maybe even, the Abort button should be called "Cancel" or "Close", because there is nothing really ongoing to abort, as far as I understand. I am not sure if the "Continue" button is indeed wired correctly to the continue action and I am not sure if I can tell the difference even if I click it.

4)
Go to:
Services>HAProxy>Settings>"Virtual Services" dropdown>Public Services
Open and existing public service (aka forntend).
The "Advertise Protocols (ALPN)" and "Select Rules" options hold entries that are sortable with drag and drop. Go to one with more than two options and try to sort. There is an annoying offset between the dragged item and your mouse:
![image](https://user-images.githubusercontent.com/8676903/82726717-9b4ebe80-9cee-11ea-8395-e7e7f8a661e3.png)
The mouse is where the red mark is.
Initially, I thought that drag and drop did not work there due to the weirdness of the situation.
I think I got some weird behavior after sorting as well, where the order was not preserved, but I can't reproduce this anymore, so maybe I can skip this.

5)
I find that there are many tooltips missing in your UI, that could explain what each button does to the user. E.g. those buttons:
![image](https://user-images.githubusercontent.com/8676903/82727010-693e5c00-9cf0-11ea-945d-939c4e916f64.png)
![image](https://user-images.githubusercontent.com/8676903/82726909-d6052680-9cef-11ea-9cd3-0e2adaca2845.png)
I had to take a look at the DOM to try and figure out what those buttons do.

6)
This should probably be posted in another place, but I find this UI a bit confusing at first glance:
Go to the Dashboard and in "Services".
![image](https://user-images.githubusercontent.com/8676903/82727106-08635380-9cf1-11ea-8aed-a64e258faf8d.png)
I think it would be more clear and easy to "read", if the status icon (green/red) was before the service name, so that you can immediately relate it to the service in question and the buttons to be under an "Actions" section. Now you think that the status icon is a button as well, and that's confusing.

7)
This is a bit different kind of feedback.
I think (I might be wrong) that there is some missing documentation on how things fall into place with all the OpnSense plugins. For example, locally I have installed and enabled the IDS/IPS, Tor, ACME, HAProxy.
I would like to get a clearer picture on how things are layered.
E.g. I would expect that the requests coming into the firewall first hit the firewall, then the IDS/IPS, then the HAProxy and/or the Tor service.
However, I see things that make me doubt that. E.g. I saw some IDS logs about traffic in WAN ports that I haven't opened (it was Tor traffic). Also, the Tor service seems to respond to any port I set it up, even if there are no firewall rules unblocking its port. I am not sure how exactly Tor works, but it should be still listening to a port, right? Isn't that the port that I set it up to listen to?
Maybe a diagram of the plugins and various systems and how they are (normally) piped, one after the other, would be enough, and it would be nice to have this somewhere in the management UI as a reference.

8)
This might be more related to security than UX:
Open the dashboard on two different browser tabs.
Log out from one of the tabs.
Go to the other tab and wait a bit.
After a while, you will see this in the dashboard:
![image](https://user-images.githubusercontent.com/8676903/82728309-3e580600-9cf8-11ea-9d90-2a8396129ef9.png)
Generally, injecting server-produced HTML snippets into pages is not considered very safe, as far as I know.
I would expect that the dashboard fetched the data it needed using AJAX requests and added it to the DOM after HTML-encoding it. But it seems that is not the case. Not sure what can be done for this.

9)
I believe it would be nice if the Let's Encrypt plugin allowed an automation for disabling a firewall rule, while it was updating the certificates. Right now you can only do that using schedules (schedule both the firewall rule drop and the L.E. certificate update) but it is tricky to setup (since it is done in two different places) and doesn't work for manual certificate updates.

**Environment**
OPNsense 20.7.1 (amd64, OpenSSL).
Virtualized inside VirtualBox on Windows 10.

**P.S. :**
If you approve some changes to any of the above, MAYBE I could try working on some, despite the fact that I don't have much free time.
My primary programming language is C#, but I write some PHP as well, once a year.
But I might have some trouble figuring out how the entire firewall is setup in the code.

**Comment by AdSchellevis** [comment]
This issue has been automatically timed-out (after 180 days of inactivity).

For more information about the policies for this repository,
please read https://github.com/opnsense/plugins/blob/master/CONTRIBUTING.md for further details.

If someone wants to step up and work on this issue,
just let us know, so we can reopen the issue and assign an owner to it.

**Comment by DumboJetEngine** [comment]
I can try working on some of those issues on my little spare time, but I am totally new to this project and have little idea on how things work on a grander scale on this project.
I think I am a quick learner and somewhat of a self-learner though.
I have already located some places in the code that need improvement, but I see that the add-ons use various different technologies.  E.g., the Tor plugin uses Volt for its PHP templates (I didn't even know what Volt was so far...), others seem to use plain PHP, bash scripts, etc, and I am not 100% sure yet how all those work together.
Also, I am not even sure how to test my changes. Last time I did changes to a similar project, I applied my changes by patching a running system. Probably this is the way to do things here too?
Anyway... I will leave this issue closed and take a look into these at my spare time. And I will create PR(s) if I have something good enough.
Unless of course you plan on replacing the UI of those add-ons and that's why you leave them without much polish. In that case, maybe I should wait for the newest UI.

**Comment by fraenki** [comment]
@DumboJetEngine First of all, I'd suggest that you limit the scope of your bug reports to only one issue at a time. You've reported so many issues for plugins/components, but these all have their individual maintainers, so nobody will feel responsible for it. :-/

> I am totally new to this project and have little idea on how things work on a grander scale on this project

Although some parts of the project still use a legacy codebase, all newer components are supposed to follow these guidelines:

https://docs.opnsense.org/development/architecture.html
https://docs.opnsense.org/development/examples/helloworld.html

This is possibly a good starting point. If you really want to contribute, then I'd suggest to start with very small changes. When your first change is ready, open a pull request and wait for a review. After some PRs you should feel more comfortable and could try something more complicated.

**Comment by DumboJetEngine** [comment]
Thanks a lot. :)
I will take a look at these.

I have thought about splitting the issues, but then I thought that they were small issues and that it would take me way longer to create multiple reports.
But I see your point.
Maybe next time. ;)
